### PR TITLE
Conda bot

### PR DESCRIPTION
**Pull request by Maarten-vd-Sande**
**What problem is the PR solving / What's new?**
See issue #133.

**What did change**
I added a conda-bot and let Jenkins periodically run it (every night).

**Does this need additional test(s)?**
Maybe.. We really need to make sure that we get all rules tested so nothing breaks accidentally.

### CODE REVIEW

**Comment by Maarten-vd-Sande** [comment]
@siebrenf are the tests covering enough on your part to feel confident for auto updates?

**Comment by siebrenf** [comment]
Yes!

Might I suggest to finish the blacklist PR, then pull develop into this branch and test the bot here?

**Comment by Maarten-vd-Sande** [comment]
The bot already "works", in the sense that it's just a script that I will have to make a periodic Jenkins job. When it finds an outdated package it opens a PR and that will be tested. Here is an example of when I was working on it:
https://github.com/vanheeringen-lab/snakemake-workflows/pull/164

**Comment by Maarten-vd-Sande** [comment]
And the blacklist PR is stuck until I find a solution to a weird snakemake edge case, or they fix it on their side.

File: Jenkins/conda-bot.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    pull_request_title = f'auto-update: {file.name}: '
+
+    lines = list(filter(None, file.decoded_content.decode("utf-8").split('\n')))
+    for line in lines:
+        # skip lines until the dependencies are listed
+        if not dependencies:
+            dependencies = 'dependencies:' in line
+        else:
+            # ignore lines starting with pip, deeper indented lines, and comments
+            if 'pip:' not in line and not line.startswith('    -') and not line.startswith('#'):
```
Comment (siebrenf) [inline Jenkins/conda-bot.py]: Ignore a wider range of text in env.yamls
```suggestion
            # ignore lines with pip, deeper indented lines, comments and empty lines
            if '  - ' in line and 'pip:' not in line and not line.startswith(('    -', '#', '\n')):
```

**Comment by Maarten-vd-Sande** [comment]
@siebrenf I think perhaps I fixed the Jenkins issue with not properly doing tests when files changed :tada:

**Pull request merged**